tofu/modules/il_gcc_application/main.tf

```
@@ -0,0 +1,218 @@
+      SMARTY_AUTH_TOKEN                  = "${module.secrets.secrets["smarty"].secret_arn}:auth_token"
+      MIXPANEL_API_KEY                   = "${module.secrets.secrets["mixpanel"].secret_arn}:api_key"
+      ENCRYPTION_KEY                     = "${module.secrets.secrets["google"].secret_arn}:encryption_key"
+      DATADOG_API_KEY                    = "${module.secrets.secrets["datadog"].secret_arn}:api_key"
+      DATADOG_APPLICATION_KEY            = "${module.secrets.secrets["datadog"].secret_arn}:app_key"
+      DATADOG_SESSION_REPLAY_SAMPLE_RATE = "${module.secrets.secrets["datadog"].secret_arn}:session_replay_sample_rate"
+      DATADOG_RUM_APPLICATION_ID         = "${module.secrets.secrets["datadog"].secret_arn}:rum_app_id"
+      DATADOG_RUM_CLIENT_TOKEN           = "${module.secrets.secrets["datadog"].secret_arn}:rum_client_token"
+      DATADOG_ENVIRONMENT                = "${module.secrets.secrets["datadog"].secret_arn}:environment"
+      JOBRUNR_DASHBOARD_ENABLED          = "${module.secrets.secrets["jobrunr"].secret_arn}:dashboard_enabled"
+      ENABLE_BACKGROUND_JOBS_FLAG        = "${module.secrets.secrets["jobrunr"].secret_arn}:enable_background_jobs"
+      WAIT_FOR_PROVIDER_RESPONSE         = "${module.secrets.secrets["il-gcc"].secret_arn}:wait_for_provider_response"
+      ALLOW_PROVIDER_REGISTRATION        = "${module.secrets.secrets["il-gcc"].secret_arn}:allow_provider_registration"
+      CONVERT_UPLOADS_TO_PDF             = "${module.secrets.secrets["il-gcc"].secret_arn}:convert_uploads_to_pdf"
+      SPRING_PROFILES_ACTIVE             = "${module.secrets.secrets["il-gcc"].secret_arn}:spring_profiles_active"
+      AWS_BUCKET                         = "${module.secrets.secrets["aws"].secret_arn}:aws_bucket"
+      AWS_REGION                         = "${module.secrets.secrets["aws"].secret_arn}:aws_region"
+      AWS_SECRET_KEY                     = "${module.secrets.secrets["aws"].secret_arn}:aws_secret_key"
+      AWS_ACCESS_KEY                     = "${module.secrets.secrets["aws"].secret_arn}:aws_access_key"
+    }
+}
+
+module "worker" {
+  source = "github.com/codeforamerica/tofu-modules-aws-fargate-service?ref=1.1.0"
+
+  project                = "illinois-getchildcare"
+  project_short          = "il-gcc"
+  stats_prefix           = "illinois-getchildcare/qa"
+  environment            = var.environment
+  service                = "worker"
+  service_short          = "worker"
+  vpc_id                 = var.vpc_id
+  private_subnets        = var.private_subnets
+  logging_key_id         = var.logging_key
+  force_delete           = var.force_delete
+  enable_execute_command = true
+  create_endpoint        = false
+  create_repository      = false
+  container_command      = ["./script/worker", "run"]
+  image_url              = module.service.repository_url
+  image_tag              = data.aws_ssm_parameter.version.insecure_value
```

If you choose to go with the `create_version_parameter` option above, you can use the output from that module here.

```suggestion
  aws_ssm_parameter      = module.application.version_parameter
```

If you choose to keep your current parameter, you can change this to use it like you would above.

```suggestion
  aws_ssm_parameter      = aws_ssm_parameter.version.name
```

---

See above comment 😅